File: GridView/FilterinIngIndicatorsInGridView/FilterinIngIndicatorsInGridView/MyGridFilterRowBehavior.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Telerik.WinControls.UI;

namespace FilterinIngIndicatorsInGridView
{
    #region CustomGridFilterRowBehavior
    public class MyGridFilterRowBehavior : GridFilterRowBehavior
    {
        protected override bool OnMouseDownLeft(MouseEventArgs e)
        {
            MyGridFilterCellElement filterCell = this.GetCellAtPoint(e.Location) as MyGridFilterCellElement;
            if (filterCell != null && filterCell.ClearButtonElement.ControlBoundingRectangle.Contains(e.Location))
            {
                return false;
            }

            return base.OnMouseDownLeft(e);
        }
    }
    #endregion
}

File: GridView/FilterinIngIndicatorsInGridView/FilterinIngIndicatorsInGridViewVB/MyGridFilterRowBehavior.vb
Imports Telerik.WinControls.UI

#Region "CustomGridFilterRowBehavior"
Public Class MyGridFilterRowBehavior
    Inherits GridFilterRowBehavior

    Protected Overrides Function OnMouseDownLeft(ByVal e As MouseEventArgs) As Boolean
        Dim filterCell As MyGridFilterCellElement = TryCast(Me.GetCellAtPoint(e.Location), MyGridFilterCellElement)

        If filterCell IsNot Nothing AndAlso filterCell.ClearButtonElement.ControlBoundingRectangle.Contains(e.Location) Then
            Return False
        End If

        Return MyBase.OnMouseDownLeft(e)
    End Function
End Class
#End Region